src/Symfony/Component/Lock/Store/RetryTillSaveStore.php

```diff
@@ -59,6 +68,10 @@ public function save(Key $key)
      */
     public function waitAndSave(Key $key)
     {
+        if (!$this->decorated instanceof BlockingStoreInterface || !$this->decorated->supportsWaitAndSave()) {
```

Comment:
What if `$this->decorated` is an implementation of the old `StoreInterface` that implements `waitAndSave()`?

Reply:
we could add method_exists() ? wdyt @jderusse ?

Reply:
Indeed, this would be a BC break.

@Simperfit you could just check if the decorator implements the `StoreInterface`
```
if (!$this->decorated instanceof BlockingStoreInterface && (!$this->decorated instanceof BlockingStoreInterface || !$this->decorated->supportsWaitAndSave()))
```
sadly, all our stores implements that method :/ making them "compatible" with `waitAndSave`

Maybe checking also if the Decorated service is not in the Symfony's Namespace... 

```php
function supportsWaitAndSave() {
  if ($this->decorated instanceof BlockingStoreInterface) {
    return $this->decorated->supportsWaitAndSave());
  }
  return ($this->decorated instanceof StoreInterface && strpos(get_class($this->decorated), __NAMESPACE__) !== 0);
}
```

Reply:
That's break the tests where we check if we can call supportsX method since it StoreInterface extends PersistStoreInterface :/